test/mux.sh

```diff
@@ -331,40 +348,78 @@ fpod=$( get_running_pod fluentd )
 wait_for_fluentd_ready
 wait_for_fluentd_to_catch_up
 
+ES_HOST_BAK=$( oc set env --list dc/logging-mux | grep \^ES_HOST= )
+OPS_HOST_BAK=$( oc set env --list dc/logging-mux | grep \^OPS_HOST= )
+
 if [ "$MUX_FILE_BUFFER_STORAGE_TYPE" = "pvc" -o "$MUX_FILE_BUFFER_STORAGE_TYPE" = "hostmount" ]; then
     os::log::info "------- Test case FILE_BUFFER_STORAGE_TYPE: $MUX_FILE_BUFFER_STORAGE_TYPE -------"
 
     update_current_fluentd $ENABLE_SECURE_FORWARD
 
-    ES_HOST_BAK=$( oc set env --list dc/logging-mux | grep \^ES_HOST= )
-    OPS_HOST_BAK=$( oc set env --list dc/logging-mux | grep \^OPS_HOST= )
-
-    # set ES_HOST and OPS_HOST to bogus
-    reset_ES_HOST ES_HOST=bogus OPS_HOST=bogus
+    # set ES_HOST and OPS_HOST to non-existing hostname
+    if [ "$ops_cluster" = "true" ]; then
+        reset_ES_HOST ES_HOST=bogus OPS_HOST=bogus_ops
+    else
+        reset_ES_HOST ES_HOST=bogus OPS_HOST=bogus
+    fi
+    os::cmd::try_until_failure 'oc get pods | egrep "logging-mux-.*-deploy"'
+    os::cmd::try_until_failure 'oc get pods | egrep "logging-mux-.*Terminating"'
+    os::cmd::try_until_text "oc get pods -l component=mux" "Running"
 
     uuid_es=$( uuidgen )
     uuid_es_ops=$( uuidgen )
 
-    logger -i -p local6.info -t $uuid_es_ops $uuid_es_ops
     add_test_message $uuid_es
+    logger -i -p local6.info -t $uuid_es_ops $uuid_es_ops
 
-    # wait long enough to make the test messages are in the buffer
-    sleep 10
-    muxpod=$( oc get pods -l component=mux -o name | awk -F'/' '{print $2}' )
+    # wait for the test messages are in the buffer
+    muxpod=$( get_running_pod mux )
+    if [ "$ops_cluster" = "true" ]; then
+       ops_filename="output-es-ops-config.output_ops_tag"
+    else
+       ops_filename="output-es-config.output_tag"
+    fi
+    retry="true"
+    while [ "$retry" = "true" ]; do
+        ops_logs=$( oc exec $muxpod -- ls /var/lib/fluentd/ | egrep $ops_filename ) || :
+        if [ -z "$ops_logs" ]; then
```

Comment:
do you need a `sleep 1` here to avoid a too tight loop?

Reply:
A good idea.  Adding it.